### DISCUSSION TITLE
Some clarification regarding the meaning of n_train, n_val and batch_size

### DISCUSSION THREAD

**Discussion by rdguha1995** [mir-group/nequip#375] [Q&A]
Hey Team,

This might be a pretty basic question. I am training an MLIP using the `extxyz` data of ~187 structures obtained from AIMD. It is a small molecule in a solvent environment of water.

My question is what does `n_train`,` n_val` and `batch_size` actually mean in terms of the training process. My naive understanding is if `n_train` is 100 and `n_val` is 50 with a default `batch_size` of 5, it will randomly take 100 structures from the training data and batch it into 20 batches of 5. Is that correct?

**Comment by Linux-cpp-lisp** [comment]
Hi @rdguha1995 ,

Thanks for your interest in our code!

Yes, that is correct, though it is worth mentioning that the splitting into batches is re-randomized at every epoch. The train-validation split is of course only randomized once at the beginning of training in this default case.
